pkg/dmaasbackup/schedule.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2020 The MayaData Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dmaasbackup
+
+import (
+	"time"
+
+	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	velerobuilder "github.com/vmware-tanzu/velero/pkg/builder"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/mayadata-io/dmaas-operator/pkg/apis/mayadata.io/v1alpha1"
+	"github.com/pkg/errors"
+	"github.com/robfig/cron"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+const (
+	// dmaasScheduleSuffix suffix for schedule name generated by dmaas operator
+	dmaasScheduleSuffix = "dmaas"
+)
+
+func (d *dmaasBackup) processSchedule(obj *v1alpha1.DMaaSBackup) error {
+	// check if crontime is mentioned
+	if obj.Spec.PeriodicFullBackupCfg.CronTime == "" {
+		return d.processNonFullBackupSchedule(obj)
```

Comment:
only difference between periodic and non-periodic is to check whether if there is due to create schedule for periodic case
by handling non full backup schedule in a different method completely, it leads to writing duplicate code in creating schedules/backup and status updates
Will the below makes sense:
- is it 'due' to create schedule/backup if periodic full backup is set?
- if no due, just return
- create the schedule/backup
- set the status related fields
- return

Reply:
We are doing the same.